find-cursor.c

```diff
@@ -212,6 +212,36 @@ int main(int argc, char* argv[]) {
 	XCloseDisplay(display);
 }
 
+// On multiscreen systems, identify which one the cursor is on
+int pointer_screen(Display *display) {
+	int screencount = ScreenCount(display);
+
+	// The traditional case
+	if (screencount == 1) {
+		return DefaultScreen(display);
+	}
+
+	// Multihead
+	for (int s=0; s < screencount; s++) {
+		int x = 0, y = 0;
+		Window window = 0;
+		Window root = 0;
+		int dummy_int = 0;
+		unsigned int dummy_uint = 0;
+		Screen *screen = ScreenOfDisplay(display, s);
+
+		int found = XQueryPointer(display, RootWindowOfScreen(screen),
+			&root, &window,
+			&x, &y, &dummy_int, &dummy_int, &dummy_uint);
+		if (found) {
+			return s;
+		}
+	}
+
+	// Fall through (should never happen)
+	return DefaultScreen(display);
```

Comment:
Maybe this should error out, or at least print a warning to stderr? Otherwise if the code fails and this "should never happen" happens anyway it will silently do the wrong thing, which be hard to spot/debug.

Reply:
I struggled over that! My thinking was that an error message would never be seen by anyone; but if this were ever to happen, the end user would notice (and report? fix?) highlights in the wrong place. But having no action at all would, I think, be puzzling. I think a warning makes sense, though -- stand by for push.